### PR TITLE
ExpressionParser: adjust triple handling for Swift REPL on Linux

### PR DESCRIPTION
**Pull request by compnerd**
The OS version is not normally part of the triple on Linux, and will
unnecessarily hamper the ability to use the Swift standard library as
the kernel version changes out-of-sync with the userspace.  The version
information on Linux is not particularly useful, and the binary
compatibility guarantees of the Linux kernel make it unnecessary.

On Windows, the version information is normally associated to the
MSVC toolchain version, not the OS version.  Furthermore, Windows has
strict guarantees about backwards guarantees, making the test less
useful.

This adjusts the triple computation to limit versioning to Darwin
platforms.

### CODE REVIEW

**Comment by compnerd** [comment]
CC: @JDevlieghere

**Comment by JDevlieghere** [comment]
Looks reasonable to me. WDYT @adrian-prantl?

File: lldb/source/Utility/ArchSpec.cpp [hunk 1]
```diff
@@ -1426,7 +1426,8 @@ bool ArchSpec::IsFullySpecifiedTriple() const {
     if ((user_specified_triple.getVendor() != llvm::Triple::UnknownVendor) ||
         TripleVendorWasSpecified()) {
       const unsigned unspecified = 0;
-      if (user_specified_triple.getOSMajorVersion() != unspecified) {
+      if (!user_specified_triple.isOSDarwin() ||
+          user_specified_triple.getOSMajorVersion() != unspecified) {
```
Comment (adrian-prantl) [inline lldb/source/Utility/ArchSpec.cpp:1430]: Could you please make this change upstream with a unit test? ArchSpec is super sensitive and I don't want any divergencies from llvm.org if we can at all avoid it.

**Review by adrian-prantl — APPROVED** [review]
Other than that this looks good.

File: lldb/source/Utility/ArchSpec.cpp (hunk 1 above)
Comment (compnerd) [reply lldb/source/Utility/ArchSpec.cpp:1430]: Sure, I can do a change upstream as well (but Id like to get this in here first to get the unified layout unblocked).

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by adrian-prantl** [comment]
Yep, as long as the upstreaming *does* happen, we can land this first.

**Pull request merged**